src/exit_handler/src/exit_handler_intel_x64_eapis_cr_access_hooks.cpp

```diff
@@ -0,0 +1,271 @@
+            return m_state_save->r13;
+
+        case general_purpose_register::r14:
+            return m_state_save->r14;
+
+        case general_purpose_register::r15:
+            return m_state_save->r15;
+    }
+
+    return 0; //Could never reach here; only for clang-tidy checks.
+}
+
+// This function is the parallel set function to get_gpr_value_by_index_reg;
+// read it's documentation to better understand it.
+void exit_handler_intel_x64_eapis::set_gpr_value_by_index_reg(uint64_t index_gpr, vmcs::value_type new_value)
+{
+    using namespace exit_qualification::control_register_access;
+    using namespace access_type;
+    using namespace general_purpose_register;
+
+    switch (index_gpr)
+    {
+        case general_purpose_register::rax:
+            m_state_save->rax = new_value;
+            break;
+
+        case general_purpose_register::rcx:
+            m_state_save->rcx = new_value;
+            break;
+
+        case general_purpose_register::rdx:
+            m_state_save->rdx = new_value;
+            break;
+
+        case general_purpose_register::rbx:
+            m_state_save->rbx = new_value;
+            break;
+
+        case general_purpose_register::rsp:
+            m_state_save->rsp = new_value;
+            break;
+
+        case general_purpose_register::rbp:
+            m_state_save->rbp = new_value;
+            break;
+
+        case general_purpose_register::rsi:
+            m_state_save->rsi = new_value;
+            break;
+
+        case general_purpose_register::rdi:
+            m_state_save->rdi = new_value;
+            break;
+
+        case general_purpose_register::r8:
+            m_state_save->r08 = new_value;
+            break;
+
+        case general_purpose_register::r9:
+            m_state_save->r09 = new_value;
+            break;
+
+        case general_purpose_register::r10:
+            m_state_save->r10 = new_value;
+            break;
+
+        case general_purpose_register::r11:
+            m_state_save->r11 = new_value;
+            break;
+
+        case general_purpose_register::r12:
+            m_state_save->r12 = new_value;
+            break;
+
+        case general_purpose_register::r13:
+            m_state_save->r13 = new_value;
+            break;
+
+        case general_purpose_register::r14:
+            m_state_save->r14 = new_value;
+            break;
+
+        case general_purpose_register::r15:
+            m_state_save->r15 = new_value;
+            break;
+    }
+}
+
+void
+exit_handler_intel_x64_eapis::handle_exit__ctl_reg_access()
+{
+    using namespace exit_qualification::control_register_access;
+    using namespace access_type;
+    using namespace general_purpose_register;
+
+    auto cr = control_register_number::get();
+    auto type = access_type::get();
+    auto index_gpr = general_purpose_register::get();
+
+    switch (cr)
+    {
+        case 0:
+        {
+            // Only cr0 loads can cause VM exits; thus no need to check type.
```

Comment:
Please remove comments, and just create a single comment block that explains the gist.